projects/Mallard/android/app/src/main/AndroidManifest.xml

```diff
@@ -4,13 +4,75 @@
+    <uses-permission android:name="${applicationId}.permission.C2D_MESSAGE" />
+    <!-- < Only if you're using GCM or localNotificationSchedule() > -->
+
+    <uses-permission android:name="android.permission.VIBRATE" />
+    <uses-permission android:name="android.permission.RECEIVE_BOOT_COMPLETED"/>
+
     <application
       android:name=".MainApplication"
       android:label="@string/app_name"
       android:icon="@mipmap/ic_launcher"
       android:roundIcon="@mipmap/ic_launcher_round"
       android:allowBackup="false"
       android:theme="@style/AppTheme">
+
+
+    <meta-data  android:name="com.dieam.reactnativepushnotification.notification_channel_name"
+            android:value="YOUR NOTIFICATION CHANNEL NAME"/>
+    <meta-data  android:name="com.dieam.reactnativepushnotification.notification_channel_description"
+                android:value="YOUR NOTIFICATION CHANNEL DESCRIPTION"/>
+    <!-- Change the resource name to your App's accent color - or any other color you want -->
+    <meta-data  android:name="com.dieam.reactnativepushnotification.notification_color"
+                android:resource="@android:color/white"/>
+
+    <!-- < Only if you're using GCM or localNotificationSchedule() > -->
+    <receiver
+        android:name="com.google.android.gms.gcm.GcmReceiver"
+        android:exported="true"
+        android:permission="com.google.android.c2dm.permission.SEND" >
+        <intent-filter>
+            <action android:name="com.google.android.c2dm.intent.RECEIVE" />
+            <category android:name="${applicationId}" />
+        </intent-filter>
+    </receiver>
+    <!-- < Only if you're using GCM or localNotificationSchedule() > -->
+
+    <receiver android:name="com.dieam.reactnativepushnotification.modules.RNPushNotificationPublisher" />
+    <receiver android:name="com.dieam.reactnativepushnotification.modules.RNPushNotificationBootEventReceiver">
+        <intent-filter>
+            <action android:name="android.intent.action.BOOT_COMPLETED" />
+        </intent-filter>
+    </receiver>
+    <service android:name="com.dieam.reactnativepushnotification.modules.RNPushNotificationRegistrationService"/>
+
+    <!-- < Only if you're using GCM or localNotificationSchedule() > -->
+    <service
+        android:name="com.dieam.reactnativepushnotification.modules.RNPushNotificationListenerServiceGcm"
+        android:exported="false" >
+        <intent-filter>
+            <action android:name="com.google.android.c2dm.intent.RECEIVE" />
+        </intent-filter>
+    </service>
+    <!-- </ Only if you're using GCM or localNotificationSchedule() > -->
+
+    <!-- < Else > -->
+    <service
+        android:name="com.dieam.reactnativepushnotification.modules.RNPushNotificationListenerService"
+        android:exported="false" >
+        <intent-filter>
+            <action android:name="com.google.firebase.MESSAGING_EVENT" />
+        </intent-filter>
+    </service>
+    <!-- </Else> -->
```

Comment:
Refer to documentation of this `reactnativepushnotification` thing, but it looks like some of these blocks should be deleted.

Reply:
Because of the FCM dependency, GCM was needed. So despite what the docs says, all the blocks are there. I left the comments in, but its probably confusing.